x-pack/platform/plugins/shared/stack_connectors/server/connector_types/slack_api/service.ts

```diff
@@ -165,46 +169,73 @@ export const createExternalService = (
     }
   };
 
-  const getChannelToUse = ({
+  const validateChannels = ({
     channels,
+    allowedList,
+  }: {
+    channels?: string[];
+    allowedList?: string[];
+  }) => {
+    if (!channels || !channels.length || !allowedList || !allowedList.length) return;
+
+    const normalizeChannel = (name: string) => name.replace(/^#/, '');
+
+    const hasDisallowedChannel = channels?.some(
+      (name) => !allowedList.some((allowed) => normalizeChannel(allowed) === normalizeChannel(name))
+    );
+
+    if (hasDisallowedChannel) {
+      throw new Error(
+        `One or more provided channel names are not included in the allowed channels list`
+      );
+    }
+  };
+
+  const getChannelToUse = ({
+    channels = [],
     channelIds = [],
+    channelNames = [],
   }: {
     channels?: string[];
     channelIds?: string[];
+    channelNames?: string[];
   }): string => {
-    if (
-      channelIds.length > 0 &&
-      allowedChannelIds &&
-      allowedChannelIds.length > 0 &&
-      !channelIds.every((cId) => allowedChannelIds.includes(cId))
-    ) {
-      throw new Error(
-        `One of channel ids "${channelIds.join()}" is not included in the allowed channels list "${allowedChannelIds.join()}"`
-      );
+    const hasChannels = channelNames.length > 0 || channelIds.length > 0 || channels.length > 0;
+
+    if (!hasChannels) {
+      throw new Error(`The channel is empty`);
```

Comment:
nit: here and below, `The channel is empty` is not very descriptive. Why not keep the old error message?